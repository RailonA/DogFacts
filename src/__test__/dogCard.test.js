import React from 'react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';
import DogCard from '../Components/dogCard';

describe('DogCard DOM', () => {
  it('renders correctly', () => {
    const dogCard = renderer.create(
      <DogCard
        name="name"
        image="dogImage"
        weight="5"
        height="5"
      />,
    ).toJSON();
    expect(dogCard).toMatchSnapshot();
  });

  it('renders card content correctly', () => {
    const dogCard = renderer.create(
      <DogCard
        name="name"
        image="dogImage"
        weight="5"
        height="5"
      />,
    ).toJSON();
    expect(dogCard.name).not.toBe('name');
  });
});
