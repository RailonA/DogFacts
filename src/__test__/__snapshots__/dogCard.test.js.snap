// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DogCard DOM renders correctly 1`] = `
<div
  className="dogCard p-3 col"
>
  <div
    className="d-flex justify-content-center "
  >
    <img
      alt="dogImage"
      className="dogImg d-flex align-self-center"
      src="dogImage"
    />
  </div>
  <div
    className="mt-3 dogMainInfo d-flex flex-column justify-content-center "
  >
    <div
      className="d-flex justify-content-center mt-3 "
    >
      <p
        className="d-flex  dogInfo mr-3"
      >
        Name: 
      </p>
      <p>
        name
      </p>
    </div>
    <div
      className="d-flex justify-content-center"
    >
      <p
        className="d-flex dogInfo mr-3"
      >
        weight:
      </p>
      <p>
        5
      </p>
      <p
        className="ml-1"
      >
        lbs
      </p>
    </div>
    <div
      className="d-flex justify-content-center"
    >
      <p
        className="d-flex dogInfo mr-3"
      >
        height:
      </p>
      <p>
        5
      </p>
      <p
        className="ml-1"
      >
        inch
      </p>
    </div>
  </div>
</div>
`;
